main.go

```diff
@@ -338,8 +338,9 @@ func Run(ctx context.Context) {
 			logger.Log.Sync()
 		}
 		stats.StopRuntimeStats()
-
-		os.Exit(1)
+		if config.GetEnv("RUDDER_GRACEFUL_SHUTDOWN", "true") == "true" {
```

Comment:
Not to confuse this with enabling or disable graceful timeout:
```suggestion
		if config.GetEnv("RUDDER_GRACEFUL_SHUTDOWN_TIMEOUT_EXIT", "true") == "true" {
```